### PR TITLE
[Tabs] First component iteration

### LINKED ISSUE #275: [Tabs] Create a Tabs component
As a page author, I want to structure content within tabs, so that I can structure the content displayed to the visitors into sections.

**Initial Requirements**

* As a page author, capability to add, remove, reorder containers
* As a page author, capability to switch to a specific container to edit its content
* As a page author, capability to edit the container title that is to be displayed in the Tab title
* The Tab should be one component with an edit config to switch between them
* In preview mode or on publish, capability to switch between the tabs

### PR DESCRIPTION
**Pull request by richardhand**
First version of a tabs switcher container component. See [0] for a feature breakdown. Implementation re-uses the UI components (Dialog and Edit Toolbar) that were introduced in the Carousel sandbox component.

[0] - https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/blob/feature/tabs/content/src/content/jcr_root/apps/core/wcm/sandbox/components/tabs/v1/tabs/README.md

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #275 
| Patch: Bug Fix?          |
| Minor: New Feature?      | 👍
| Major: Breaking Change?  |
| Tests Added + Pass?      | 👍
| Documentation Provided   | 👍 (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

**Comment by richardhand** [comment]
@vladbailescu - addressed your feedback in https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/pull/284/commits/1958daab4e0737bb97c8f7ce4bae51bda0a19403, https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/pull/284/commits/aff4e7d417dfe10588447faa96a7bcf1db46cc88

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @Self
+    private SlingHttpServletRequest request;
+
+    private List<ListItem> items;
+
+    private List<ListItem> readItems() {
+        List<ListItem> items = new ArrayList<>();
+        if (resource != null) {
+            ComponentManager componentManager = request.getResourceResolver().adaptTo(ComponentManager.class);
+            for (Resource res : resource.getChildren()) {
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java]: I would add a null check on `componentManager`.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import com.adobe.cq.wcm.core.components.models.ListItem;
+import com.adobe.cq.wcm.core.components.sandbox.models.Container;
+import com.day.cq.wcm.api.components.Component;
+import com.day.cq.wcm.api.components.ComponentManager;
+
+/**
+ * Abstract class which can be used as base class for {@link Container} implementations.
+ */
+public abstract class AbstractContainerImpl implements Container {
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java]: What are the reasons for having an `abstract` class? We could have a Container model one day and `ContainerImpl` would be its impl.

File: testing/it/ui-js/src/content/jcr_root/apps/core/wcm/sandbox/tests/test-cases/Tabs/v1/Tabs.js [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+            .execTestCase(tabs.tcCreateItems(selectors))
+
+            // switch to the content frame and focus the first tab
+            .config.changeContext(c.getContentFrame)
+            .click(selectors.tabs.tab + ":first-child")
+
+            // simulate a right arrow keydown event and verify right navigation
+            .simulate(selectors.tabs.tab + ":focus", "keydown", { keyCode: keyCodes.ARROW_RIGHT })
+            .asserts.isTrue(function() {
+                var $tabActive = h.find(selectors.tabs.tabActive + ":contains('item1')");
```
Comment (jckautzmann) [inline testing/it/ui-js/src/content/jcr_root/apps/core/wcm/sandbox/tests/test-cases/Tabs/v1/Tabs.js]: We should also test that the right tabpanel is active.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java (hunk 1 above)
Comment (jckautzmann) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java]: I added it in commit: https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/pull/284/commits/4be968ae0d6f3218ff8ba199ea5ebb228450d14d

Comment (richardhand) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java]: Thanks!

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java (hunk 2 above)
Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java]: We talked about this offline as well, I actually like the way things are structured currently. I believe we won't have a basic container in the future but rather something smarter, which might include other information (like the current responsive grid); its interface might inherit from `Container` and its implementation from `AbstractContainerImpl`. So it makes sense to keep what we have. We can refactor when we feel it makes sense as it's all implementation details.

Comment (jckautzmann) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/AbstractContainerImpl.java]: +1

**Pull request merged**